docs/requirements/hardware-proposal.md

```diff
@@ -0,0 +1,98 @@
+# Hardware proposal
+
+Current document covers requirements in proposal real hardware solutions,
+which meet requirements gathered earlier.
+
+## Microchip ATTPM20P
+
+* Link to documentation: [mcp_attpm20p](https://ww1.microchip.com/downloads/en/DeviceDoc/ATTPM20P-Trusted-Platform-Module-TPM-2.0-SPI-Interface-Summary-Data-Sheet-DS40002082A.pdf)
+* Compliant to the Trusted Computing Group (TCG) Trusted Platform Module (TPM)
+Version 2.0, r116 Trusted Platform Module Library
+* Single-Chip Turnkey Solution
+* Hardware Asymmetric Crypto Engine
+* Microchip ARM® M0+Microprocessor
+* Internal FLASH Storage for Keys
+* Serial Peripheral Interface (SPI) Protocol up to 36 MHz
+* Secure Hardware and Firmware Design and Device Layout
+* FIPS-140-2 Module Compliant Including the High-Quality Random Number
+Generator (RNG), HMAC, AES, SHA, ECC, and RSA Engines
+
+* Offered in Commercial (0°C to +70°C) Temperature Range
+  * Supply Voltage: 1.8V to 3.3V
+* Offered in Industrial (-40°C to +85°C) Temperature Range
+  * Supply Voltage: 3.3V
+* Cryptographic Support for:
+  * HMAC
+  * AES-128
+  * SHA-1
+  * SHA-256
+  * ECC BN_P256, ECCNIST_P256
+  * RSA 1024-2048 bit keys
+* 16 KB of User-Accessible Nonvolatile Memory
+* X.509 EK Certificates (Optional)
+
+## Infineon CY8C6xxxx Series
+
+* Link to documentation: [Infineon PSoC6 Family](https://www.infineon.com/cms/en/product/microcontroller/32-bit-psoc-arm-cortex-microcontroller/psoc-6-32-bit-arm-cortex-m4-mcu/)
+* Point of interest: PSoC 64-Secured MCU series
+* 32-bit Programmable System-on-Chip, integrates in one structure
+ARM Cortex M4 (150MHz) with Cortex M0+, 100MHz
+* Various variants of memories:
+  * flash 384-1856 KB
+  * RAM 176-944KB
+* Interfaces I2C, SPI, UART, USB
+* Crypto Accelerator (DES/TDES, AES, SHA, CRC, TRNG, RSA/ECC)
+* Software ModusToolbox: [Modus Toolbox](https://www.infineon.com/cms/en/design-support/tools/sdk/modustoolbox-software/)
+
+## SparkFun QuickLogic Thing Plus - EOS S3
+
+* Link to documentation: [QuickLogic Thing Plus - EOS S3 Datasheet](https://www.sparkfun.com/products/17273)
+* eFPGA-enabled ARM  Cortex©-M4F MCU
+* Fully supported with Zephyr RTOS, FreeRTOS
+* Based fully on open source hardware
+* Software environment is round 100% open source
+(with support for FreeRTOS, Zephyr, nMingen, Docker, and SymbiFlow)
```

Comment:
FreeRTOS does not support it, now. (current info, taken from: https://www.freertos.org/RTOS_ports.html , seems to be valid).
About Zephyr info as above.

Reply:
From [wikipedia](https://en.wikipedia.org/wiki/Open-source_hardware):

> The term usually means that information about the hardware is easily discerned so that others can make it – coupling it closely to the [maker movement](https://en.wikipedia.org/wiki/Maker_culture).[[1]](https://en.wikipedia.org/wiki/Open-source_hardware#cite_note-1) Hardware design (i.e. mechanical drawings, [schematics](https://en.wikipedia.org/wiki/Circuit_diagram), [bills of material](https://en.wikipedia.org/wiki/Bills_of_material), [PCB](https://en.wikipedia.org/wiki/Printed_circuit_board) layout data, [HDL](https://en.wikipedia.org/wiki/Hardware_description_language) source code[[2]](https://en.wikipedia.org/wiki/Open-source_hardware#cite_note-freehw-2) and [integrated circuit](https://en.wikipedia.org/wiki/Integrated_circuit) layout data), in addition to the software that [drives](https://en.wikipedia.org/wiki/Device_driver) the hardware, are all released under free/[libre](https://en.wikipedia.org/wiki/Libre_content) terms.

I'm not sure whether EOS S3 can be considered fully Open Source - Open Source tools lack BlockRAM support, which supported only by https://www.quicklogic.com/products/fpga/fpga-development-tools/, QuickWorks is not open source. @krystian-hebel thoughts? BTW I find an old [PR for yosys](https://github.com/YosysHQ/yosys/pull/1816) which adds support for EOS S3, maybe we could find some use of it?